contributing.md

```diff
@@ -1,9 +1,126 @@
+
+Here are links to all the sections in this document:
+
+<!-- If you change any of the headings in this document, remember to update the table of contents. -->
+
+- [Code of Conduct](#code-of-conduct)
+- [Technologies Used](#technologies-used)
+- [Development Installation](#development-installation)
+- [File Structure](#file-structure)
+- [New Design](#new-design)
+- [How Can I Contribute ?](#how-can-i-contribute?)
+  - [First Timers](#first-timers)
+  - [Want something more challenging](#want-something-more-challenging)
+  - [Feature Enhancement](#feature-enhancement)
+- [Creating a Pull request](#creating-a-pull-request)
+  - - [Tips](#tips)
+
+# Code of Conduct
+
+Please follow the guidelines mentioned at [CODE OF CONDUCT.md](https://github.com/processing/p5.js-web-editor/blob/master/CODE_OF_CONDUCT.md#p5js-code-of-conduct).
+
+# Technologies Used
+
+ > **MERN stack - MongoDB, Express, React/Redux, and Node**. 
+ 
+ - For a reference to the **file structure format** I am using, please look at the [Mern Starter](https://github.com/Hashnode/mern-starter).
+
+ - This project **does not use CSS Modules, but uses Sass**. [BEM guidelines](http://getbem.com/) are followed. 
+ 
+ - For CSS naming conventions, write OOSCSS (Object-Oriented SCSS) with placeholders and mixins, and follow the [7-1 Pattern](https://sass-guidelin.es/#the-7-1-pattern) for Sass.
+
+ - We're using [ES6](http://es6-features.org/) and transpiling to ES5 using [Babel](https://babeljs.io/). 
+
+ - For reference to the JavaScript style guide, see the [Airbnb Style Guide](https://github.com/airbnb/javascript), [React ESLint Plugin](https://github.com/yannickcr/eslint-plugin-react).
+
+ - The ESLint configuration is based on a few popular React/Redux boilerplates. Open to suggestions on this. If in development, **you're getting annoyed with ESLint**, you can **remove** it from `webpack.config.dev.js` in the JavaScript loader, or disable any line from eslint by commenting at the end of the line `// eslint-disable-line`.
+
+# Development Installation
+
+To get started on a patch, first read the instruction from [README.md](https://github.com/processing/p5.js-web-editor#development-installation).
+
+# File Structure
+
+```
+/client
+|-- /components
+|   |-- accountNav.jsx
+|   |-- forceProtocol.jsx
+|   |-- Nav.jsx
+|-- /images //InlineSVG files present    
+|-- /modules
+    |-- /App
+    |-- /IDE
+    |-- /User
+```
+
+# New Design
+
+Design proposed and theme changes are present at: [Zeplin](https://scene.zeplin.io/project/55f746c54a02e1e50e0632c3).
+
+# How Can I Contribute?
+
+### First Timers
+For first-time contributors or those who want to start with a small task: [check out our list of good first bugs](https://github.com/processing/p5.js-web-editor/labels/good%20first%20issue). These issues have an assigned mentor to help you out and are great issues for learning about the web editor, its file structure and our development process.
```

Comment:
There's not an "assigned mentor" for these tasks but someone will definitely help out (probably me since I'm currently the lead maintainer). I think it would also be good to add a sentence in here that's like "It's okay to not know how to fix an issue and feel free to ask questions about to approach the problem! We are all just here to learn and make something awesome."